### PR TITLE
New definition of "H", "L", "M", "T", new feature `FORMS` (if needed)

### PR DESCRIPTION
**Pull request by Clayblockunova**
afaik from MDN, `document.forms+[]` returns to "[object HTMLCollection]" in most browsers, but idk how this actually work in old Android webview, so I created a new feature just in case (idk how to simulate, tho). btw, afaik, `forms` have a relatively shorter code in those similar properties of `document`, but idk is that the shortest.

### CODE REVIEW

**Comment by Clayblockunova** [comment]
what should I do?

**Comment by fasttime** [comment]
Thanks! I just need some time to look into this.

**Comment by fasttime** [comment]
It looks like there are already better definitions for all of the characters. Am I missing something?

**Comment by Clayblockunova** [comment]
> It looks like there are already better definitions for all of the characters. Am I missing something?

you mean there is something better than `forms`?

**Comment by fasttime** [comment]
I think so. At least I could not find a case where the new definitions produce shorter code for some combination of browsers.

**Comment by Clayblockunova** [comment]
> I think so. At least I could not find a case where the new definitions produce shorter code for some combination of browsers.

the expected combination is browsers except old Android web view.

**Comment by fasttime** [comment]
`FORMS` is available in Android Browser 4.0 and 4.1, but we still need the feature because it's not available in Node.js.

**Comment by Clayblockunova** [comment]
can we just treat that's part of `ANY_DOCUMENT`?

**Comment by fasttime** [comment]
> can we just treat that's part of `ANY_DOCUMENT`?

I think it's easier to treat it as a separate feature. If we make `ANY_DOCUMENT` more specific we will also need to redefine `DOCUMENT`, and the emulation logic for both features.

**Comment by Clayblockunova** [comment]
and, the "V" definition based on `document.createElement` relies on `ANY_DOCUMENT` now. should we create a separate feature for this?

**Comment by Clayblockunova** [comment]
or should I add new simulation now? what should I do?

**Comment by fasttime** [comment]
> and, the "V" definition based on `document.createElement` relies on `ANY_DOCUMENT` now. should we create a separate feature for this?

Yes, that would be better. I will split a new feature.

File: src/lib/features.js [hunk 1]
```diff
@@ -361,6 +361,21 @@ var featureInfos =
             return available;
         },
     },
+    FORMS:
+    {
+        description:
+        'Existence of the global object document having the property forms, which have string ' +
+        'representation [object HTMLCollection]".',
```
Comment (fasttime) [inline src/lib/features.js:368]:
```suggestion
        'Existence of the object document.forms with string representation "[object ' +
        'HTMLCollection]".',
```

File: src/lib/features.js [hunk 2]
```diff
@@ -361,6 +361,21 @@ var featureInfos =
...
+        check:
+        function ()
+        {
+            var available = typeof document.forms === 'object' &&
+            document.forms + '' === '[object HTMLCollection]';
```
Comment (fasttime) [inline src/lib/features.js:373]:
```suggestion
            var available =
            typeof document === 'object' && document.forms + '' === '[object HTMLCollection]';
```

`document` is not defined in Node.js.

**Comment by Clayblockunova** [comment]
And, how to sumulate?

File: src/lib/features.js [hunk 3]
```diff
@@ -361,6 +361,21 @@ var featureInfos =
...
+        'HTMLCollection]".',
+        check:
+        function ()
+        {
+            var available =
+            typeof document === 'object' && document.forms + '' === '[object HTMLCollection]';
+            return available;
+        },
+        includes:   ['ANY_DOCUMENT'],
+        attributes: { 'web-worker': 'web-worker-restriction' },
```
Comment (fasttime) [inline src/lib/features.js:377]: I've changed my mind. Let's just use `ANY_DOCUMENT` for all `document` related features, but let us keep `FORMS` as an alias for `ANY_DOCUMENT` when `document.forms` is required in a definition.

```suggestion
        aliasFor: 'ANY_DOCUMENT',
```

The other changes in this file can be reverted.

**Comment by fasttime** [comment]
> And, how to sumulate?

To emulate `document.forms` we can add a new property to the stub for `document` in the function `makeEmuFeatureDocument`:

https://github.com/fasttime/JScrewIt/blob/05243b9554e94f48764679238a383e27970be87a/test/helpers/feature-emulation.helpers.js#L224

for example:

```js
                document =
                {
                    createElement:  createElement,
                    forms:          '[object HTMLCollection]',
                    nodeName:       '#document',
                };
```

**Review by fasttime — APPROVED** [review]
Nice work, thanks!

**Pull request merged**